File: server/src/main/java/net/progruzovik/dissent/config/WebSocketConfig.java
package net.progruzovik.dissent.config;

import net.progruzovik.dissent.socket.DissentHandshakeInterceptor;
import net.progruzovik.dissent.socket.WebSocketHandler;
import org.springframework.context.annotation.Configuration;
import org.springframework.lang.NonNull;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
public class WebSocketConfig implements WebSocketConfigurer {

    private final WebSocketHandler webSocketHandler;
    private final DissentHandshakeInterceptor handshakeInterceptor;

    public WebSocketConfig(WebSocketHandler webSocketHandler, DissentHandshakeInterceptor handshakeInterceptor) {
        this.webSocketHandler = webSocketHandler;
        this.handshakeInterceptor = handshakeInterceptor;
    }

    @Override
    public void registerWebSocketHandlers(@NonNull WebSocketHandlerRegistry registry) {
        registry.addHandler(webSocketHandler, "/app/")
                .addInterceptors(handshakeInterceptor)
                .setAllowedOrigins("*");
    }
}

File: server/src/main/kotlin/net/progruzovik/dissent/config/WebSocketConfig.kt
package net.progruzovik.dissent.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.handler.SimpleUrlHandlerMapping
import org.springframework.web.reactive.socket.WebSocketHandler
import org.springframework.web.reactive.socket.server.support.WebSocketHandlerAdapter

@Configuration
class WebSocketConfig(private val webSocketHandler: WebSocketHandler) {

    @Bean
    fun handlerMapping() = SimpleUrlHandlerMapping().apply {
        order = 1
        urlMap = mapOf("/app" to webSocketHandler)
    }

    @Bean
    fun handlerAdapter() = WebSocketHandlerAdapter()
}
